### ISSUE TITLE
namedtuple does not accept f strings as string literal for the first arg

### ISSUE THREAD

**Issue by cy** [python/mypy#9046]
When creating a namedtuple, if the first argument is an f string, mypy throws `main.py:3: error: namedtuple() expects a string literal as the first argument`

In https://mypy-play.net/?mypy=latest&python=3.8
```python
from collections import namedtuple
a = "b"
test = namedtuple(f"{a}", ["a"])
```
throws
```
main.py:3: error: namedtuple() expects a string literal as the first argument
Found 1 error in 1 file (checked 1 source file)
```
Interestingly, if string interpolation is *not* used inside the fstring, then mypy does not complain:
```python
from collections import namedtuple
test = namedtuple(f"b", ["a"])
```
returns
```
Success: no issues found in 1 source file
```

As https://www.python.org/dev/peps/pep-0498/ considers fstrings string literals, I'd expect fstrings to be accepted as the first arg to namedtuple.

I suspect this bug is caused by this line https://github.com/python/mypy/blob/c1f1f9566feefa2ade91b880f731d0b0cf76ff09/mypy/semanal_namedtuple.py#L276 where an fstring is surprisingly not a StrExpr, BytesExpr or UnicodeExpr:
(ipython)
```python
In [1]: b = "a"

In [2]: a = f"{b}"

In [8]: from mypy.nodes import (StrExpr, BytesExpr, UnicodeExpr)

In [10]: any(isinstance(a, expr) for expr in [StrExpr, BytesExpr, UnicodeExpr])
Out[10]: False
```

Locally I am using mypy 0.770 and python 3.8; but in mypy-play.net I observe the same thing happening on mypy latest.

**Comment by JelleZijlstra** [comment]
We could make this work in the specific example you gave, where the f-string consists only of a single literal string interpolated into it, but I'm not sure why that would be useful: you can just write `namedtuple(a, ["a"])`, which should work once `a` is declared `Final`.

But in general, mypy doesn't support dynamically creating namedtuple names or members, and that's unlikely to change.

**Comment by cy** [comment]
I'm disappointed, but I see. Thanks for your response.
